[VisualStudioMarketplace] Add support to prerelease extensions version (Issue #8207)

OronW opened the pull request:
Added pre-release version support to VScode marketplace service.
Updated response schema.
Updated Tests to include pre-release version.

PR fixing issue #8207

chris48s commented on services/visual-studio-marketplace/visual-studio-marketplace-version.service.js:
```diff
@@ -27,15 +41,22 @@ export default class VisualStudioMarketplaceVersion extends VisualStudioMarketpl
+  transform({ json }, includePrereleases) {
     const { extension } = this.transformExtension({ json })
-    const version = extension.versions[0].version
+    const preRelease = 'Microsoft.VisualStudio.Code.PreRelease'
```
I think the intent here would be a bit clearer if we called this var `preReleaseKey` or `preReleaseTag`

OronW replied:
I agree. Fixed.

chris48s commented on services/visual-studio-marketplace/visual-studio-marketplace-version.service.js:
```diff
@@ -27,15 +41,22 @@ export default class VisualStudioMarketplaceVersion extends VisualStudioMarketpl
     const { extension } = this.transformExtension({ json })
-    const version = extension.versions[0].version
+    const preRelease = 'Microsoft.VisualStudio.Code.PreRelease'
+    const version = includePrereleases
+      ? extension.versions[0].version
+      : extension.versions.find(
+          obj => !obj.properties.find(({ key }) => key === preRelease)
+        ).version
+
```
There are a couple of problems here.

1. If there are only pre-releases and no stable releases this will fail because `.find()` will just return `undefined`. For this case, we should fall back to showing the latest pre-release if there are no stable releases at all (this is what we normally do for other badges). 
2. This one is a bit more of an edge case: It looks like the deal is pre-releases are indicated by `{"key": "Microsoft.VisualStudio.Code.PreRelease", "value": "true"}` and stable releases are indicated by complete absence of that object, but I think if we were to encounter a release with `{"key": "Microsoft.VisualStudio.Code.PreRelease", "value": "false"}` we should also consider that stable (whether or not we currently think this is likely). This code would classify that as a pre-release.

chris48s reviewed (commented):
Thanks for submitting this and including tests. It is looking in pretty good shape to start with and its clear you've worked through the documentation and contributing guidelines. :+1:

chris48s commented:
Minor point: I've edited the PR title so it will run the tests for the whole VSCode Marketplace service family on the next push. This PR modifies the `VisualStudioMarketplaceBase` class which impacts on all the badges that inherit from it.

OronW commented:
@chris48s 
Thank you for the review and all your comments!

I've updated my code accordingly, and I've added 2 more tests - 
1. For the case where there are only prereleases but the include_prereleases flag is false (fallback to the latest version)
2. For the case where we might get a "false" value in the prerelease object.

I've also run the entire VisualStudioMarketplace tests locally without issues.

Please let me know if there's anything else :)

chris48s reviewed (approved):
great first contribution :+1:
